gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -113,32 +109,63 @@ impl GatewayLdkClient {
         })?;
 
         let (htlc_stream_sender, htlc_stream_receiver) = tokio::sync::mpsc::channel(1024);
+        let task_group = TaskGroup::new();
+
+        // LDK Node will sync the wallet's chain automatically in the background, but in
+        // the background it only sync every 10 seconds. In devimint we want
+        // this to sync faster, so we spawn a thread that manually syncs the chain.
+        if is_env_var_set(FM_IN_DEVIMINT_ENV) {
+            let node_clone = node.clone();
+            task_group.spawn("ldk node sync event handler", |handle| async move {
+                loop {
+                    if handle.is_shutting_down() {
+                        return;
+                    }
```

Comment:
You can just spawn the task as cancellable imo, then the `is_shutting_down` check becomes redundant.